craft_archives/repo/apt_key_manager.py

```diff
@@ -87,53 +119,48 @@ def get_key_fingerprints(cls, *, key: str) -> List[str]:
 
         :returns: List of key fingerprints/IDs.
         """
-        # pyright: reportUnknownMemberType=false, reportUnknownVariableType=false
-        with tempfile.NamedTemporaryFile(suffix="keyring") as temp_file:
-            return (  # type: ignore
-                gnupg.GPG(keyring=temp_file.name).import_keys(key_data=key).fingerprints
-            )
+        with tempfile.NamedTemporaryFile() as temp_file:
+            temp_file.write(key.encode())
+            temp_file.flush()
+
+            response = _call_gpg("--show-keys", temp_file.name).splitlines()
```

Comment:
I see that `_call_gpg` has provisions to get input from stdin, I wonder if it would be easy to just pipe the key here instead of creating a temporary file.

Reply:
You're right! While I was at it I noticed that we should be using `--with-colons` for some of the calls to gpg, so I added that here too.